aten/src/ATen/BatchedFallback.cpp

```diff
@@ -195,6 +195,33 @@ void batchedTensorInplaceForLoopFallback(const c10::OperatorHandle& op, torch::j
   torch::jit::push(stack, self);
 }
 
+static Tensor safeStack(TensorList tensors) {
+  auto is_defined = [](const Tensor& t) { return t.defined(); };
+  if (std::all_of(tensors.begin(), tensors.end(), is_defined)) {
+    return at::stack(tensors);
+  }
+  // NOTE [vmap through backward and undefined grad]
+  // While vmapping through backward functions (to compute batched grad), it
+  // is possible for the backward function to return an undefined grad for some
+  // grad_input for each example. In that case, we return an undefined grad.
+  //
+  // It is theoretically posssible for *some* of the examples to produce an
+  // undefined grad (a kernel could peek at the gradient values and return an
+  // undefined tensor if it determines the gradient is full of zeros). We
+  // could handle this by treating the undefined grad as a zero-filled tensor
+  // of the correct shape while stacking the tensors together. However I expect
+  // this to happen very rarely (I have not been able to find an example in our
+  // codebase) so we just error out in this case.
+  if (std::none_of(tensors.begin(), tensors.end(), is_defined)) {
+    return Tensor();
+  }
+  TORCH_CHECK(false,
+      "vmap: slow fallback received a mix of undefined and defined tensors ",
+      "as the result of an operation. This is not supported. If this error ",
+      "message appears in framework code (e.g. autograd), please file us ",
```

Comment:
nit: maybe we could encourage the user to always report an issue on github? I am very curious what they would do to hit that :p

Reply:
One thing a user could do is have a forward function that they created via c++ extensions that sometimes returns undefined tensors... but yeah, would be curious about the use case for anything like that